src/app/zone-ingresses/data/index.spec.ts

```diff
@@ -0,0 +1,287 @@
+import { describe, expect, test } from 'vitest'
+
+import { ZoneIngressOverview } from './'
+import { useResponder } from '@/test-support'
+import mock from '@/test-support/mocks/src/zone-ingresses/_/_overview'
+import { ZoneIngressOverview as PartialZoneIngressOverview } from '@/types/index'
+
+type Writeable<T> = { -readonly [P in keyof T]: Writeable<T[P]> };
+
+type Assert = (actual: ZoneIngressOverview, item: PartialZoneIngressOverview) => void
+type Context = {
+  message: string
+  assert: Assert
+  get: () => Promise<PartialZoneIngressOverview>
+}
+type Test = {
+  env?: Record<string, string>
+  message: string
+  assert: Assert
+  setup: (item: PartialZoneIngressOverview) => Promise<PartialZoneIngressOverview>
+}
+
+const freeze = (obj: any) => {
+  Object.keys(obj).forEach(prop => {
+    if (typeof obj[prop] === 'object' && !Object.isFrozen(obj[prop])) {
+      freeze(obj[prop])
+    }
+  })
+  return Object.freeze(obj)
+}
+
+const get = async (env: Record<string, string>) => {
+  const responder = useResponder({
+    _: mock,
+  }, (key: string, d = '') => env[key] ?? d)
+  const request = responder('_')
```

Comment:
**Question**: What’s up with this underscore value?

Reply:
Yeah little weird I know, the responder requires a URL, normally these are like `/meshes/_/dataplanes/_`

```javascript
responder('_') // <--- this
_: mock, <-- looks up this
```

In this case if we wanted to be absolutely correct, it would be `/zones`, but for testing purposes we really don't care what the url is, plus this is hidden from the user and means this function is reusable with less options to pass in.

Reply:
Also just noticed, please be sure not to use this file for code review, its for comparison with what I had previously (unPRed) only